pandas/io/html.py

```diff
@@ -197,36 +199,39 @@ def parse_tables(self):
         tables = self._parse_tables(self._build_doc(), self.match, self.attrs)
         return (self._build_table(table) for table in tables)
 
-    def _parse_raw_data(self, rows):
-        """Parse the raw data into a list of lists.
+    def _parse_tables(self, doc, match, attrs):
+        """Return all tables from the parsed DOM.
 
         Parameters
         ----------
-        rows : iterable of node-like
-            A list of row elements.
+        doc : parser object
```

Comment:
I think you still need to supply the actual data type of the input i.e. what is the output of type(`doc`) when used in this function?  Just add a line in the function that prints out the `type` and recompile `pandas` to include it.  Then see what it is and then add that to the documentation.

Reply:
OK. It depends on what parser you use, though (BeautifulSoup or lxml). It's the BS4 or lxml parser object. I will check exact types.

Reply:
Arghhh!  I wish you had told me this sooner. :cry:

Reply:
For future reference, we are *generally* very picky about the data types in the doc-string.  However, as the data-type is dependent on an external factor (i.e. engine), we drop the data type and immediately dive into the description of the variable.

Reply:
Here is what I mean by that for example:

BEFORE
~~~python
row : node-like
     A row in the HTML table
~~~

AFTER
~~~python
row : a row in the HTML table
~~~

Reply:
Sorry. Didn't know that until I dived into the documentizing. So right now I am consistent about this kind of docstring:

```
        rows : list of HTML row elements
            A list of row elements of a table, usually <tr> or <th> elements.
```

but the `row element` here is also dependent on the parser (it's either a BS4 or lxml row element). This does seem like the right way to describe it though. (I left the documentation structured as is in the way it described subclassed functions; there is an virtual function, and that's what's documented, and then subclassed functions specific to the parser that are not documented.)

So: 

- Would you like me to back out the changes I made (the ones like the above) and redo them as you suggest?
- Based on what you suggest, `parser object` does seem like the right way to describe it ...?

Reply:
@jowens : Instead of providing a "data-type" like `parser object`, go right into describing the data input, so in the example you provided above, do this:

~~~
rows : a list of row elements of a table, usually <tr> or <th> elements.
~~~

Reply:
OK. For where I have "parser object", here's the actual type: `<class 'bs4.BeautifulSoup'>`. Suggestion for what to put instead of "parser object"?

Reply:
Just do this:

~~~python
doc : the DOM from which to parse the table element.
~~~

Reply:
Many thanks. Next round pushed. You are a very patient person.